sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/LogicalRelation.scala

```diff
@@ -102,3 +109,18 @@ object LogicalRelation {
     LogicalRelation(relation, toAttributes(schema), Some(table), false)
   }
 }
+
+/**
+ * Extract the [[BaseRelation]] and [[CatalogTable]] from [[LogicalRelation]]. This also provides
+ * [[LogicalRelation]] itself, to cover the case where other params are needed (but not required
+ * to have specific value to match).
+ */
+object RelationAndCatalogTable {
```

Comment:
`LogicalRelationWithTable`?

Reply:
Don't have a strong voice for the name; it looks better coupled with your above comment. Will rename.